tools/extract_scripts.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+
+# these two pages have some relevant information:
+# https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions
+# https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners
+
+import argparse
+import re
+import sys
+from pathlib import Path
+from typing import Any, Dict
+
+import yaml
+
+Job = Dict[str, Any]
+
+windows_labels = {'windows-latest', 'windows-2019'}
```

Comment:
this might not always be true when we get windows self hosted runners, might need to make this a bit more broad

Reply:
Agreed, but I didn't know how to future-proof this since I don't know what naming scheme we plan to use for our Windows self-hosted runners; should I change this to just check whether the label starts with `'windows'`?

Also, this isn't foolproof anyways, since it doesn't account for the case where e.g. the value of `runs-on` is an `${{ <expression> }}` that uses something from the `matrix`. I figured that I can just list GitHub's built-in Windows labels for now, and then we can change this later if we need to.

Reply:
Can't we check for `shell` specification on the job rather than rely on labels? (Also, because default shell can change in the future, unless specified, see https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell

Reply:
We can, but that would require us to explicitly specify `shell` everywhere in our workflows. Is that something we want to do?